lib/livebook/config.ex

```diff
@@ -514,18 +537,20 @@ defmodule Livebook.Config do
   Parses zero trust identity provider from env.
   """
   def identity_provider!(env) do
-    case System.get_env(env) do
-      "googleiap:" <> rest ->
-        {Livebook.ZTA.GoogleIAP, rest}
-
-      "cloudflare:" <> rest ->
-        {Livebook.ZTA.Cloudflare, rest}
-
-      nil ->
-        nil
+    if provider = System.get_env(env) do
+      identity_provider!(env, provider)
+    end
+  end
 
-      _ ->
-        abort!("invalid configuration for identity provider")
+  @doc """
+  Parses and validates zero trust identity provider within context.
+  """
+  def identity_provider!(context, provider) do
+    with [type, key] <- String.split(provider, ":", parts: 2),
```

Comment:
Btw is it always a single key by definition, or do we expect a more elaborate config in the future?

Reply:
We don't know. If it gets complex in the future, then we could delegate this to the module somehow.